src/openai/azure/_async_client.py

```diff
@@ -378,6 +379,25 @@ def auth_headers(self) -> Dict[str, str]:
             return { 'Authorization': f'Bearer {self.credential.get_token()}'}
         return {"api-key": self.api_key}
 
+    def _prepare_request(self, request: httpx.Request) -> None:
+        # TODO: need confirmation that it is okay to override this
+        url = request.url
+        if url.path.startswith("/audio"):
+            model_name = request.stream.fields[0].value  # is this a robust way to extract model name?
+            request.url = url.copy_with(path=f"/openai/deployments/{model_name}{url.path}")
+            return
```

Comment:
Can't get model out of the `content` here b/c I don't want to call read() on the request before I send it..... it's probably not a  a robust way to extract the model name from the httpx._multipart.MultipartStream. Need to re-think this approach...